Code/Tools/ProjectManager/Platform/Mac/O3DE_SDK_Launcher.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzFramework/Process/ProcessWatcher.h>
+
+#include <cstdlib>
+#include <mach-o/dyld.h>
+
+int main(int argc, char* argv[])
+{
+    // We need to pass in the engine path since we won't be able to find it by searching upwards.
+    // We can't use any containers that use our custom allocator till after the call to ComponentApplication::Create()
+    char enginePathParam[AZ_MAX_PATH_LEN];
+    char processPathStr[AZ_MAX_PATH_LEN];
+    uint32_t pathLen = AZ_MAX_PATH_LEN;
+    _NSGetExecutablePath(processPathStr, &pathLen);
+    snprintf(enginePathParam, AZ_MAX_PATH_LEN, "--engine-path=%s/../Engine", processPathStr);
+
+    char *commandLineParams[AZ_MAX_PATH_LEN] = {processPathStr, enginePathParam, nullptr};
```

Comment:
Suggestion here is to use `AZStd::fixed_vector<char*, 3>` with an AZ::IO::FixedMaxPath for storing the path data for long enough to make it to the ComponentApplication constructor call

```suggestion
    AZ::IO::FixedMaxPath processPathStr;
    uint32_t pathLen = static_cast<uint32_t>(processPathStr.capacity());
    _NSGetExecutablePath(processPathStr.Native().data(), &pathLen);
    AZ::IO::FixedMaxPath enginePathParam = processPathStr / ".."/ "Engine";

    // Uses the fixed_vector deduction guide to determine the type is AZStd::fixed_vector<char*, 3>
    AZStd::fixed_vector commandLineParams{ processPathStr.data(), enginePathParam.data(), static_cast<char*>(nullptr) };
```

Reply:
Trying this out.

Reply:
Fixed